src/service_aggregator.py

```diff
@@ -678,13 +679,71 @@ async def multi_strider(messages, params, guid):
     return response, status_code
 
 
-def merge_answer(results, qnode_ids):
-    # We are going rename most of the node and edge bindings.  But, we want to preserve bindings to things in the
-    # original query.  For the current creative one-hop, that is just nodes b/c we are replacing the one edge.
-    # The original qnode bindings are not necessarily the same because of the dumb way that we are handling
-    # subclass entailment.  If one of the merged answers involved a subclass, then the qnode binding will be
-    # to the subclass. So we loop over all the results and find all the different qnode bindings by serializing.
-    mergedresult = {"node_bindings": {}, "edge_bindings": {}}
+def create_aux_graph(analysis):
+    """Given an analysis, create an auxiliary graph.
+    Look through the analysis edge bindings, get all the knowledge edges, and put them in an aux graph.
+    Give it a random uuid as an id."""
+    aux_graph_id = str(uuid.uuid4())
+    aux_graph = { "edges": [] }
+    for edge_id, edgelist in analysis["edge_bindings"].items():
+        for edge in edgelist:
+            aux_graph["edges"].append(edge["id"])
+    return aux_graph_id, aux_graph
+
+
+def add_knowledge_edge(result_message, aux_graph_ids, answer):
+    """Create a new knowledge edge in the result message, with the aux graph ids as support."""
+    # Find the subject, object, and predicate of the original query
+    query_graph = result_message["message"]["query_graph"]
+    #get the first key and value from the edges
+    qedge_id, qedge = next(iter(query_graph["edges"].items()))
+    #For the nodes, if there is an id, then use it in the knowledge edge. If there is not, then use the answer
+    qnode_subject_id = qedge["subject"]
+    qnode_object_id = qedge["object"]
+    if "ids" in query_graph["nodes"][qnode_subject_id] and query_graph["nodes"][qnode_subject_id]["ids"] is not None:
+        qnode_subject = query_graph["nodes"][qnode_subject_id]["ids"][0]
+        qnode_object = answer
+    else:
+        qnode_subject = answer
+        qnode_object = query_graph["nodes"][qnode_object_id]["ids"][0]
+    predicate = qedge["predicates"][0]
+    # Create a new knowledge edge
+    new_edge_id = str(uuid.uuid4())
+    new_edge = {
+        "subject": qnode_subject,
+        "object": qnode_object,
+        "predicate": predicate,
+        "attributes": [
+            {
+                "attribute_type_id": "support graph",
+                "value": aux_graph_ids
+            }
+        ],
+        "sources": [{"resource_id":"infores:aragorn", "resource_role":"biolink:primary_knowledge_source"}]
```

Comment:
Just double checking that, in this context, Aragorn is a primary_knowledge_source

Reply:
This is the expected behavior

Reply:
Good double check, the answer is yes, but I'll add a comment as to why.